### DISCUSSION TITLE
Defects (or facts I feel uncomfortable) of current lazer mania editor

### DISCUSSION THREAD

**Discussion by middlered** [ppy/osu#29859] [Ideas]
There are not so many feedbacks on the mania editor so I'll just leave some subjective ones there.  
_All things are taken from the current version of the game._ `Lazer 2024.906.2`

### 1. Distracting divisor line hiding effect
![image](https://github.com/user-attachments/assets/2582badf-e99c-451d-af59-a66627f5fcce)
Imagine when you only want to focus on the notes and patterns but the line is flashing when you move your mouse and distracting you.   
Also the divisor line should not be hidden when playing in the editor. It is crucial for mania mappers when checking the map. Providing an option for not hiding divisor line (and set it as default) is necessary.

### 2. Note's bottom cover the divisor line
| In Stable | In Lazer |
| - | - |
| ![image](https://github.com/user-attachments/assets/2ef6b112-5d72-47e7-98e7-3d045a2e4d5b) | ![image](https://github.com/user-attachments/assets/476979b1-436b-4838-96f1-abb66355f0e6) |

Again, divisor line is crucial.

### 3. Bring bookmarker back
Already have one discussion about it: https://github.com/ppy/osu/discussions/29169
And do not forget `Ctrl+B` hotkey for creating it.

### 4. Kiai indicator
| In Stable | In Lazer |
| - | - |
| ![image](https://github.com/user-attachments/assets/f3f77329-4055-474a-a838-d5a2187c7ea3) | huh |

### 5. Millisecond
This is not just for mania only, sometimes ppl may want to do some math things on calculating timing (like making SV maps in mania) and I believe it is helpful for providing a view of the current millisecond in the bottom left corner.
| In Lazer | 
| - | 
| ![image](https://github.com/user-attachments/assets/99f46cf1-021c-48ff-b928-418126b5e008) | 

### 6. Test gameplay stat
| In Stable| 
| - | 
| ![image](https://github.com/user-attachments/assets/e981b668-4c5c-47de-9d69-8f1085f4fbca) | 

It is useful. Bring it back plz.

### 7. Top timeline
| In Lazer | 
| - | 
| ![image](https://github.com/user-attachments/assets/abf46bc6-87df-4ba6-99d7-a6926a427ec5) | 

Tbh, objects on timeline is not only useless in mania. Mania playfield already provided a fairly clean presentation of notes and I don't see any reason we need objects on timeline, it however reduces the height of the editor playfield, leaving such little vertical space for a VSRG gamemod makes me feel really uncomfortable.  
Of course, we can make a compromise solution by placing the timeline beside the editor playfield and making it vertical, instead of on the top. I think this is better.

### 9. Playing behaviour
| In Stable | In Lazer |
| - | - |
| ![image](https://github.com/user-attachments/assets/09afd7ec-3fbb-42ec-9d2c-fa251f92af1d) | ![image](https://github.com/user-attachments/assets/8426238c-837f-497d-9ddd-2447f97e90f7) |

Mania map is heavily focused on context consistency, don't clean or block the past notes. Also I think it is possible to provide an option to disable hit lights in editor to reduce distraction.

### 13. Edit/Gameplay skin
For mania, skins for gameplay are fairly different from default skin, such as removal of LN head or thicker note. These skins are suitable for gameplay but not editing map. Make an opinion for skin used in editor to separate from gameplay skin is a good idea.

### 14. Reference
| In Stable | 
| - | 
| ![image](https://github.com/user-attachments/assets/5e0be300-059c-48d8-85e0-a8aabbe5d44d) | 

Don't need much to say. It is necessary.

---
Some are solved

#### Density graph
Bring it back plz. 

####  Background movement
The background behavior is the same in song selection and editor. I don't need background wiggle with my mouse movement when I'm mapping. This is distracting.

####  Counter-intuitive mouse pointer
When moving the note, the mouse pointer may rotate, which makes me feel 'unsteady' and very counter-intuitive to the direction I expected to move.

#### LN placement
When placing long notes, the editor will automatically jump to the place where LN started. It is an awful experience when making LN-oriented maps. Make an option to disable it.

**Comment by bdach** [comment]
> This is not just for mania only, sometimes ppl may want to do some math things on calculating timing (like making SV maps in mania) and I believe it is helpful for providing a view of the current millisecond in the bottom left corner.

I don't understand this one, what do you want specifically? The time in milliseconds? Stable doesn't show that either does it?

I don't think density graph is happening anymore, I've seen it be referred to as "not useful" or even "so bad" in internal discussion.

> When placing long notes, the editor will automatically jump to the place where LN started. It is an awful experience when making LN-oriented maps. Make an option to disable it.

There is one already. See view menu in editor.

> When moving the note, the mouse pointer may rotate, which makes me feel 'unsteady' and very counter-intuitive to the direction I expected to move.

There is an option in game settings to turn this off.

> The background behavior is the same in song selection and editor. I don't need background wiggle with my mouse movement when I'm mapping. This is distracting.

There is an option in game settings to turn this off.

**Reply by middlered** [reply]
> I don't understand this one, what do you want specifically? The time in milliseconds? Stable doesn't show that either does it?

Afaik the only way to directly retrieve the current time in the form of milliseconds is by creating a new timing point and copying its offset value both in stable and lazer. It would be great if the game could directly give out the current offset in the compose view.
![image](https://github.com/user-attachments/assets/0def5553-f5a4-4d7b-acee-e21337f034aa)

> I don't think density graph is happening anymore, I've seen it be referred to as "not useful" or even "so bad" in internal discussion.

okey but there are [tools](https://github.com/sinanates17/Huggeds-Mania-Spread-Wizard) that compare spread density for modding. The density graph is definitely still useful, it just needs some improvements.

**Reply by bdach** [reply]
> okey but there are tools that compare spread density for modding. The density graph is definitely still useful, it just needs some improvements.

That can be part of the new "checks" thing, which is aimod v2 / mapsetverifier integrated into the game.

**Comment by peppy** [comment]
> Also the bar line should not be hidden when playing in the editor

Do you mean they are disappearing because too thin (ie. probably monitor-can't-display-properly) or is there something more to this?

**Reply by middlered** [reply]
um specifically I mean the the divisor line. It disappears if you don't hover over the playfield in note mode and not gonna appear in select mode.

**Reply by peppy** [reply]
Right, understood. Currently only show in placement mode.
